enhancements/update/reduced_reboots.md

```diff
@@ -0,0 +1,328 @@
+high should document those findings.
+
+Thus far RHCOS, Node, MCO, SDN, Containers, and PSAP teams are known to fall into
+this group of teams which have components coupled to the host.
+
+#### MCO - Widen Node Constraints to allow for N-2
+
+Building upon the EUS-to-EUS upgrade MVP work(https://github.com/openshift/enhancements/blob/master/enhancements/update/eus-upgrades-mvp.md#mco---enforce-openshifts-defined-host-component-version-skew-policies)
+to allow MCO to enforce host constraints we will broaden those constraints to
+enable host component version skew.
+
+Admins who choose to would then be able to skip a host reboot by following this
+pattern:
+
+1. Starting with a 4.8 cluster, pause Worker MachineConfigPool
+1. Upgrade to 4.9
+1. Upgrade to 4.10
+1. Unpause Worker MachineConfigPool
+
+Note that this can be decoupled in a way that when we ship 4.9 the initial MCO
+could assert constraints which require 4.9 host components before upgrading to
+4.10. Then after we ship 4.10 and have sufficiently tested a 4.8 to 4.10 host
+component version skew a later 4.9.z MCO could have its constraints broadened.
+This allows us additional time to test broader version skews if we so choose.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This imposes significant risk due to a number of factors:
+- We're currently not confident in upstream's testing matrix and our specific
+  feature sets
+- We've never before expected teams to offer broader than N-1 compatibility.
+  Teams have always assumed at most N-1 and even then, especially early after a
+  GA release, it's not uncommon to find problems in N-1 compatibility.
+- While N-1 is tested in the context of upgrades it's not been tested in long
+  term use.
+- If external integrations depend on host components of all nodes having been
+  updated then we'll run into problems. For instance, if there's an upgrade
+  scenario where RHV cloud provider integration needs to be upgraded between 4.6
+  and 4.10 in order to ensure compatibility and the components which interface
+  with RHV are components of RHCOS then we may not upgrade those components at
+  the same minor version expected previously.
+
+We may mitigate some of this by further delaying EUS-to-EUS upgades after normal
+minor version upgrades have been promoted to stable and allocating significantly
+more time and effort to testing efforts. Ultimately this introduces another
+dimension to an already complex testing matrix.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Should we make these between specific named versions. ie: 4.6-4.8, and 4.8-4.10
+or should this be a standard N-2 rule, ie: 4.6-4.8, 4.7-4.9, 4.8-4.10?
+
```

Comment:
What happens with an MCP which is "never" unpaused? Will MCO force an upgrade if the MCP violates the constraint N-1 or N-2

Reply:
No, we'd just inhibit upgrades that fall outside of our defined policy. That can still be forced around but anytime an admin chooses to force an upgrade they become responsible for the disaster they create.

Reply:
Never unpausing a pool will sooner or later fail the nodes out as the kubelets stop getting Kube-API CA rotations, even if the admins don't force an update.  We should be very clear that pool pauses are acceptable practice for a few hours, maybe into days, but that the whole time pools are paused, you are not getting CA rotations, bumped pull secrets, RHCOS bugfixes, and all the other good stuff that comes with having an unpaused pool.  Folks who want to leave a pool paused for longer should be considering all of those consequences, and deciding for themselves if the reduced disruption is worth the risk.